tensorrt_llm/_torch/auto_deploy/transform/library/quantize_moe.py

```diff
@@ -0,0 +1,201 @@
+                scale_nodes_group.append(scales)
+
+        return new_attrs, scale_nodes_group
+
+    # Quantize all three expert weights
+    w1_attrs, w1_scales = quantize_param_list(w1_names)
+    w2_attrs, w2_scales = quantize_param_list(w2_names)
+    w3_attrs, w3_scales = quantize_param_list(w3_names)
+
+    # Collect scale tensors per scale type across w1, w2, w3
+    def collect_scales(index: int) -> Tuple[List[Node], List[Node], List[Node]]:
+        return (
+            [s[index] for s in w1_scales],
+            [s[index] for s in w2_scales],
+            [s[index] for s in w3_scales],
+        )
+
+    # Prepare args
+    args = [
+        node.args[0],  # x
+        node.args[1],  # selected_experts
+        node.args[2],  # routing_weights
+        w1_attrs,
+        w2_attrs,
+        w3_attrs,
+    ]
+
+    for idx in range(len(scale_keys)):
+        s1, s2, s3 = collect_scales(idx)
+        args.extend([s1, s2, s3])
+
+    # Replace the current node with the quantized version
+    with gm.graph.inserting_after(node):
+        new_node = gm.graph.call_function(
+            quantized_op,
+            args=tuple(args),
+        )
+        ad_logger.debug(f"Updating {node.name} args to {new_node.args}")
+        node.replace_all_uses_with(new_node)
+        gm.graph.erase_node(node)
+
+
+# TODO(Fridah-nv): robust handling similar to `extract_param_names_from_lin_node` or expand it
+def _extract_moe_weight_param_lists(moe_node: Node) -> Tuple[List[str], List[str], List[str]]:
+    """
+    Given a torch.ops.moe.torch_moe node in gm.graph, extract three lists of
+    the parameter names for w1_weight, w2_weight, and w3_weight.
+
+    Returns:
+      (w1_names, w2_names, w3_names), each a list of strings like 'layer.expert_0.w1.weight'
+    """
+    # args layout: (x, selected_experts, routing_weights, w1_list, w2_list, w3_list)
+    try:
+        w1_list, w2_list, w3_list = moe_node.args[3:6]
+    except ValueError:
+        raise RuntimeError(
+            f"Expected moe_node.args to have at least 6 entries, got {len(moe_node.args)}"
+        )
+
+    def _unwrap_list(arg) -> List[str]:
+        if not isinstance(arg, (list, tuple)):
+            raise TypeError(f"Expected a Python list/tuple of get_attr Nodes, got {type(arg)}")
+        names: List[str] = []
+        for elt in arg:
+            if not isinstance(elt, Node) or elt.op != "get_attr":
+                raise RuntimeError(f"Expected each list element to be a get_attr Node, got {elt}")
+            names.append(elt.target)
+        return names
+
+    w1_names = _unwrap_list(w1_list)
+    w2_names = _unwrap_list(w2_list)
+    w3_names = _unwrap_list(w3_list)
+
+    return w1_names, w2_names, w3_names
+
+
+class QuantizationConfig(TransformConfig):
+    """Configuration for the quantization transform."""
+
+    quant_config: Dict[str, Any] = Field(
+        description="Quantization configuration with quant_algo and optioanlly exclude_modules"
+    )
```

Comment:
quant_config comes from the model factory. So instead of putting it into the config of the transform, you should be able to retrieve from the factory provided to `_apply`